superset/tasks/async_queries.py

```diff
@@ -76,30 +67,35 @@ def load_chart_data_into_cache(
     # pylint: disable=import-outside-toplevel
     from superset.charts.data.commands.get_data_command import ChartDataCommand
 
-    try:
-        ensure_user_is_set(job_metadata.get("user_id"))
```

Comment:
I'm not certain whether this logic was correct, i.e., should we ensure that `job_metadata.get("user_id")` is the user performing said actions regardless of whether there is a current user set? Note the `override_user` context manager will reset back to the current user upon exit.

@benjreinhart it seems you authored https://github.com/apache/superset/pull/13878 and thus I was hoping you might be able to provide more context, i.e., on line #75 should it be `override_user(user)` rather than `overide_user(user, force=False)`?